numpy/ma/core.py

```diff
@@ -5057,7 +5057,7 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=np._NoValue):
 
         if self._mask is nomask:
             result = super(MaskedArray, self).mean(axis=axis,
-                                                   dtype=dtype, **kwargs)
+                                                   dtype=dtype, **kwargs)[()]
         else:
```

Comment:
For anyone coming here in future, the above is no longer the case. `[()]` does not promote scalars to 0d-arrays